app/decorators/user_decorator.rb

```diff
@@ -51,6 +51,12 @@ def active_or_pending_profile
     user.active_profile || pending_profile
   end
 
+  # This user's most recently activated profile that has also been deactivated
+  # due to a password reset, or nil if there is no such profile
+  def password_reset_profile
+    user.profiles.password_reset.order(activated_at: :desc).first
```

Comment:
the logic here is subtly different if users have multiple profiles.

If a user has profile A first an B second, if the user resets A due to password reset, then creates B and does not reset B, this would return A when we expect it to return nil by the old logic